### DISCUSSION TITLE
would it work with ollama?

### DISCUSSION THREAD

**Discussion by fahadshery** [Pythagora-io/gpt-pilot#1111] [Q&A]
I have ollama up and running. would this work with that?

**Comment by hqnicolas** [comment]
Yes, it works with ollama
https://github.com/hqnicolas/Ollama-Pilot-CasaOs
and you can use the openai to connect over WEB to an openwebui server....
it's amazing!!!!!!!

**Comment by mercury64** [comment]
I just cloned GPT-Pilot, had Ollama previously setup.

All you have to change is change the base_url, api_key and give it a model name.

My config:
```json
  "llm": {
    "openai": {
      // Base url to the provider/server, omitting the trailing "chat/completions" part.
      "base_url": "http://localhost:11434/v1",
      "api_key": "Ollama",
      "connect_timeout": 60.0,
      "read_timeout": 20.0
    },
...
...
  "agent": {
    "default": {
      "provider": "openai",
      "model": "llama3.3",
      "temperature": 0.5
    }
  },
```

Make sure you have the v1 at the end of the base_url.
Make sure to set api_key to Ollama
To see what models you have, got to: http://localhost:11434/api/tags

I have a complex app running now, should have done something simpler, might take a while.
